### DISCUSSION TITLE
Component test with Vitest fails with autoimported composables

### DISCUSSION THREAD

**Discussion by anaroblesd** [nuxt/nuxt#19655] [Questions] [answered]
Hello, 
I've tried to test my components using vitest and I'm having problems that I can't solve with other discussion threads. 
I'm using "nuxt": "^3.2.2" with <script setup> . When running my component test, it fails because of auto imports like useFecth or useState. It is possible to configure it in any way? I'm using the `unplugin-auto-import` module too, but I can only make it work for vue imports as ref(). 
Thanks in advance. BR, Ana.

**Comment by dsvgl** [comment]
check https://github.com/danielroe/nuxt-vitest

**Comment by anaroblesd** [comment, marked as answer]
Hello, I found a solution for mocking the Nuxt auto imports on my test componets. Here there's an example: https://medium.com/@ana.robles.developer/nuxt-3-vitest-vue-test-utils-mockear-en-los-test-funciones-auto-importadas-2242297eb06d

**Reply by dsvgl** [reply]
You'd have to stub all nuxt internal composables (useAppConfig, useNuxtApp, ...) this way. I found/used this method in the beginning, too. But now, there is https://github.com/danielroe/nuxt-vitest which takes care of all the auto-imports. You can also mock Nuxt imports and components with it. Not yet used `useLazyFetch` myself, so I cannot say if this one works.
